tools/raisinbread_refresh.php

```diff
@@ -0,0 +1,286 @@
+ * file and provides the name of their LORIS database.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Loris
+ * @author   John Saigle <john.saigle@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+$info = <<<INFO
+This script is used by LORIS developers to DELETE DATA and replace
+them with new test data.
+
+
+INFO;
+
+echo $info;
+
+echo "\e[0;31m*** Executing this script will result in the LOSS OF DATA! ***\e[0m\n";
+
+$cwd = getcwd();
+if (substr_compare($cwd, 'tools', mb_strlen($cwd) - mb_strlen('tools')) !== 0) {
+    die('Please run this script from the tools/ directory.' . PHP_EOL);
+}
+
+try {
+    require_once 'generic_includes.php';
+
+    $dbInfo = $config->getSettingFromXML('database');
+
+    if (! $config->getSetting('dev')['sandbox']) {
+        throw new \LorisException(
+            "Config file indicates that this is not a sandbox. Aborting to " .
+            "prevent accidental data loss."
+        );
+    }
+    $dbname = $dbInfo['database'];
+    $host = $dbInfo['host'];
+    // The "quat" user and password should be configured to a user with DROP
+    // and CREATE permissions for the database.
+    $username = $dbInfo['quatUser'];
+    $password = $dbInfo['quatPassword'];
+
+    $urlConfigSetting = $config->getSetting('url');
+    $baseConfigSetting = $config->getSetting('base');
+    $hostConfigSetting = $config->getSetting('host');
+
+} catch (\DatabaseException $e) {
+    echo 'Could not connect to the database in the Config file.' .
+        'It\'s possible that the LORIS tables have already been dropped.' .
+        PHP_EOL;
+    echo 'Continue attempting to install Raisinbread database? (yN)' . PHP_EOL;
+    $input = trim(fgets(STDIN));
+
+    if (mb_strtolower($input) !== 'y') {
+        die;
+    }
+    echo PHP_EOL . 'Please enter the name of your database:' . PHP_EOL;
+    $dbname = trim(fgets(STDIN));
+} catch (\LorisException $e) {
+    die($e->getMessage());
+} catch (Exception $e) {
+    die("Could not load project/config.xml");
+}
+
+
+echo <<<CONFIRMATION
+Please type the database name `$dbname` to confirm you wish to drop tables
+and import test data: 
+CONFIRMATION;
+
+$input = trim(fgets(STDIN));
+if ($input !== $dbname) {
+    die('Input did not match database name. Exiting.');
+}
+
+$mysqlCommand = <<<CMD
+mysql -A $dbname
+CMD;
+
+// Test whether a connection to MySQL is possible via a config file.
+// If not, read DB information from the config file. This method is not as 
+// preferable because it generates MySQL warnings due to the password being
+// supplied via a command-line argument.
+exec($mysqlCommand . ' -e "show tables;" 2>&1 1>/dev/null', $output, $status);
+if ($status != 0) {
+    $mysqlCommand = <<<CMD
+mysql -A "$dbname" -u "$username" -h "$host" -p$password
+CMD;
+}
+
+// Drop tables
+echo PHP_EOL .'Dropping LORIS tables....' . PHP_EOL;
```

Comment:
The script outputs the following:
```
Please type the database name `loris` to confirm you wish to drop tables
and import test data: loris

Dropping LORIS tables....
...
[*] Creating LORIS core tables.... 
...
[*] Creating instrument tables.... 
...
[*] Importing Raisinbread data....
...
Restoring config setting `url` to value `http://localhost:8080`

Restoring config setting `base` to value `/var/www/loris/`

Restoring config setting `host` to value `localhost`

Please choose a new password for the admin user:
Running command: `php resetpassword.php admin`
```
However, my OCD would be very happy if headers of the different sections were consistant :): (for example:)
```
Please type the database name `loris` to confirm you wish to drop tables
and import test data: loris

[*] Dropping LORIS tables....
...

[*] Creating LORIS core tables.... 
...

[*] Creating instrument tables.... 
...

[*] Importing Raisinbread data....
...

[*] Restoring config settings...
Restoring config setting `url` to value `http://localhost:8080`
Restoring config setting `base` to value `/var/www/loris/`
Restoring config setting `host` to value `localhost`

[*] Changing the admin password...
Running command: `php resetpassword.php admin`
```
In short: always have a new line before a section, have each sections identified with `[*]` before it, no empty lines between the different restoring config setting messages, adding a section name for the change to the admin password.

Does that make sense?

Reply:
Yep, I'll do that